run/django/e2e_test.py

```diff
@@ -0,0 +1,409 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This test creates a Cloud SQL instance, a Cloud Storage bucket, associated
+# secrets, and deploys a Django service
+
+import os
+import subprocess
+import uuid
+
+import pytest
+import requests
+from google.cloud import secretmanager_v1 as sm
+
+# Unique suffix to create distinct service names
+SUFFIX = uuid.uuid4().hex[:10]
+
+PROJECT = os.environ["GOOGLE_CLOUD_PROJECT"]
+DATABASE_VERISON = "POSTGRES_12"
+REGION = "us-central1"
+DATABASE_TIER = "db-f1-micro"
+
+CLOUDSQL_INSTANCE = f"instance-{SUFFIX}"
+STORAGE_BUCKET = f"{PROJECT}-media-{SUFFIX}"
+DATABASE_NAME = f"polls-{SUFFIX}"
+DATABASE_USERNAME = f"django-{SUFFIX}"
+DATABASE_PASSWORD = uuid.uuid4().hex[:26]
+
+ADMIN_NAME = "admin"
+ADMIN_PASSWORD = uuid.uuid4().hex[:26]
+
+# These are hardcoded elsewhere in the code
+SECRET_SETTINGS_NAME = "django_settings"
+SECRET_ADMINPASS_NAME = "superuser_password"
+
+
+@pytest.fixture
+def project_number():
+    projectnum = (
+        subprocess.run(
+            [
+                "gcloud",
+                "projects",
+                "list",
+                "--filter",
+                PROJECT,
+                "--format",
+                "value(projectNumber)",
+            ],
+            stdout=subprocess.PIPE,
+            check=True,
+        )
+        .stdout.strip()
+        .decode()
+    )
+
+    yield projectnum
+
+
+@pytest.fixture
+def postgres_host(project_number):
+    # Create database instance
+    subprocess.run(
+        [
+            "gcloud",
+            "sql",
+            "instances",
+            "create",
+            CLOUDSQL_INSTANCE,
+            "--database-version",
+            DATABASE_VERISON,
+            "--tier",
+            DATABASE_TIER,
+            "--region",
+            REGION,
+            "--project",
+            PROJECT,
+        ],
+        check=True,
+    )
+
+    # Create database
+    subprocess.run(
+        [
+            "gcloud",
+            "sql",
+            "databases",
+            "create",
+            DATABASE_NAME,
+            "--instance",
+            CLOUDSQL_INSTANCE,
+            "--project",
+            PROJECT,
+        ],
+        check=True,
+    )
+    # Create User
+    # NOTE Creating a user via the tutorial method is not automatable.
+    subprocess.run(
+        [
+            "gcloud",
+            "sql",
+            "users",
+            "create",
+            DATABASE_USERNAME,
+            "--password",
+            DATABASE_PASSWORD,
+            "--instance",
+            CLOUDSQL_INSTANCE,
+            "--project",
+            PROJECT,
+        ],
+        check=True,
+    )
+
+    # Allow access to database from Cloud Build
+    subprocess.run(
```

Comment:
I'll have to talk to you more about what permanent fixtures I can setup

Reply:
This may be moot if we don't have to create the DB as part of the fixture, but having multiple subprocesses in a single fixture can lead to leftover resources that don't get cleaned up if something fails. The more decoupled they can be, the better

Reply:
Let's set this in the project. Let me know if you don't have access.